src/providers/trusted_service/context.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use log::info;
+use mbox::MBox;
+use std::ffi::{c_void, CString};
+use std::io::{Error, ErrorKind};
+use std::ptr::null_mut;
+use ts_binding::*;
+
+#[allow(
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals,
+    clippy::unseparated_literal_suffix,
+    // There is an issue where long double become u128 in extern blocks. Check this issue:
+    // https://github.com/rust-lang/rust-bindgen/issues/1549
+    improper_ctypes,
+    missing_debug_implementations,
+    trivial_casts,
+    clippy::all,
+    unused,
+    unused_qualifications
+)]
+pub mod ts_binding {
+    include!(concat!(env!("OUT_DIR"), "/ts_bindings.rs"));
+
+    unsafe impl Send for rpc_caller {}
+    unsafe impl Sync for rpc_caller {}
+
+    unsafe impl Send for service_context {}
+    unsafe impl Sync for service_context {}
+}
+
+#[allow(
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals,
+    clippy::unseparated_literal_suffix,
+    // There is an issue where long double become u128 in extern blocks. Check this issue:
+    // https://github.com/rust-lang/rust-bindgen/issues/1549
+    improper_ctypes,
+    missing_debug_implementations,
+    trivial_casts,
+    clippy::all,
+    unused,
+    unused_qualifications
+)]
+mod ts_protobuf {
+    include!(concat!(env!("OUT_DIR"), "/ts_crypto.rs"));
+}
+
+#[derive(Debug)]
+pub struct Context {
+    rpc_caller: Option<rpc_caller>,
+    service_context: Option<service_context>,
+    rpc_session_handle: Option<MBox<c_void>>,
+}
+
+impl Context {
+    pub fn connect() -> anyhow::Result<Self> {
+        info!("Querying for crypto Trusted Services");
+        let mut status = 0;
+        let service_context = unsafe {
+            service_locator_query(
+                CString::new("sn:tf.org:crypto:0").unwrap().into_raw(),
```

Comment:
Will that string always stay the same? Does it need to be part of the configuration?

Reply:
Potentially part of the configuration, but I have no clue for now, tbh. It kinda looks static to me, but who knows

Reply:
Ok we can check that one later.